src/wp-includes/class-wp-block-type.php

```diff
@@ -270,6 +287,70 @@ public function __construct( $block_type, $args = array() ) {
 		$this->set_props( $args );
 	}
 
+	/**
+	 * Proxies getting values for deprecated properties for script and style handles for backward compatibility.
+	 * Gets the value for the corresponding new property if the first item in the array provided.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $name Deprecated property name.
+	 *
+	 * @return string|null|void The value read from the new property if the first item in the array provided,
+	 *                          null when value not found, or void when unknown property name provided.
+	 */
+	public function __get( $name ) {
+		if ( ! in_array( $name, $this->deprecated_properties ) ) {
+			return;
+		}
+
+		$new_name = $name . '_handles';
+		return isset( $this->{$new_name }[0] ) ? $this->{$new_name }[0] : null;
+	}
+
+	/**
+	 * Proxies checking for deprecated properties for script and style handles for backward compatibility.
+	 * Checks whether the corresponding new property has the first item in the array provided.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $name Deprecated property name.
+	 *
+	 * @return boolean Returns true when for the new property the first item in the array exists,
+	 *                     or false otherwise.
+	 */
+	public function __isset( $name ) {
+		if ( ! in_array( $name, $this->deprecated_properties ) ) {
+			return false;
+		}
+
+		$new_name = $name . '_handles';
+		return isset( $this->{$new_name }[0] );
+	}
+
+	/**
+	 * Proxies setting values for deprecated properties for script and style handles for backward compatibility.
+	 * Sets the value for the corresponding new property as the first item in the array.
+	 * It also allows setting custom properties for backward compatibility.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $name  Property name.
+	 * @param mixed  $value Property value.
+	 */
+	public function __set( $name, $value ) {
+		if ( ! in_array( $name, $this->deprecated_properties ) ) {
+			$this->{$name} = $value;
+			return;
+		}
+
+		if ( ! is_string( $value ) ) {
+			return;
+		}
+
+		$new_name = $name . '_handles';
```

Comment:
In the future, we might want to use some sort of deprecation mechanism similar to `_deprecated_argument` for functions/methods. It would be a great opportunity to share the message with those block developers that still use `register_block_type` to pass an array with the deprecated properties instead of `block.json`.